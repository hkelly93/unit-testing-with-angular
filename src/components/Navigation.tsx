import React from 'react';

import { ThemeConsumer } from '../context/ThemeContext';

type Props = {
  forwards: () => void;
  backwards: () => void;
};

const Navigation = ({ forwards, backwards }: Props) => (
  <ThemeConsumer>
    {(theme: any) => (
      <div className={theme.navigation}>
        <button onClick={backwards} className={theme['left-arrow']}>
          {'<'}
        </button>
        <button onClick={forwards} className={theme['right-arrow']}>
          {'>'}
        </button>
      </div>
    )}
  </ThemeConsumer>
);

export default Navigation;
