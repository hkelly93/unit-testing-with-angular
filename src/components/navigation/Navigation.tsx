import React from 'react';

import { ThemeConsumer } from '../../context/ThemeContext';

type Props = {
  forwards: () => void;
  backwards: () => void;
};

export default ({ forwards, backwards }: Props) => (
  <ThemeConsumer>
    {(theme: any) => (
      <div className={theme.navigation}>
        <button onClick={backwards} className={theme['left-arrow']}>
          {'<'}
        </button>
        <button onClick={forwards} className={theme['right-arrow']}>
          {'>'}
        </button>
      </div>
    )}
  </ThemeConsumer>
);
