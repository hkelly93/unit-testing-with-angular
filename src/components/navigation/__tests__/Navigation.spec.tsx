import React from 'react';
import {
  render,
  waitForElement,
  fireEvent,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ThemeContext from '../../../context/ThemeContext';
import Navigation from '../Navigation';

describe('Navigation Component', () => {
  afterEach(cleanup);

  const props = { forwards: jest.fn(), backwards: jest.fn() };

  const getComponent = (p = props) => (
    <ThemeContext.Provider value={{}}>
      <Navigation {...p} />
    </ThemeContext.Provider>
  );

  test('loads and displays navigation arrows', () => {
    const { asFragment } = render(getComponent());

    expect(asFragment).toMatchSnapshot();
  });

  test('calls props#forwards when the right arrow is clicked', () => {
    const { getByText } = render(getComponent());

    fireEvent.click(getByText('>'));
    expect(props.forwards).toHaveBeenCalled();
  });

  test('calls props#backwards when the left arrow is clicked', () => {
    const { getByText } = render(getComponent());

    fireEvent.click(getByText('<'));
    expect(props.backwards).toHaveBeenCalled();
  });
});
