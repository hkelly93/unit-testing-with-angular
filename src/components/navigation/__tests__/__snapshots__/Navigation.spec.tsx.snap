// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component loads and displays navigation arrows 1`] = `[Function]`;
